dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.gateway
+
+import java.net.InetSocketAddress
+import java.util.Locale
+import javax.inject.{ Inject, Named }
+
+import akka.actor.{ ActorRef, ActorSystem }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.ws._
+import akka.pattern.ask
+import akka.stream.Materializer
+import akka.stream.scaladsl.{ Flow, Keep, Sink, Source }
+import akka.util.Timeout
+import com.lightbend.lagom.discovery.ServiceRegistryActor.{ Found, NotFound, Route, RouteResult }
+import com.lightbend.lagom.internal.javadsl.registry.ServiceRegistryService
+import org.slf4j.LoggerFactory
+import play.api.inject.ApplicationLifecycle
+import play.api.routing.Router.Routes
+import play.api.routing.SimpleRouter
+
+import scala.collection.immutable
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+
+class AkkaHttpServiceGatewayFactory @Inject() (lifecycle: ApplicationLifecycle, config: ServiceGatewayConfig,
+                                               @Named("serviceRegistryActor") registry: ActorRef)(implicit actorSystem: ActorSystem, mat: Materializer) {
+
+  def start(): InetSocketAddress = {
+    new AkkaHttpServiceGateway(lifecycle, config, registry).address
+  }
+}
+
+class AkkaHttpServiceGateway(lifecycle: ApplicationLifecycle, config: ServiceGatewayConfig, registry: ActorRef)(implicit actorSystem: ActorSystem, mat: Materializer) {
+
+  private val log = LoggerFactory.getLogger(classOf[AkkaHttpServiceGateway])
+
+  import actorSystem.dispatcher
+  private implicit val timeout = Timeout(5.seconds)
+  val http = Http()
+
+  private val handler = Flow[HttpRequest].mapAsync(4) { request =>
```

Comment:
Note that setting this to 1 has zero impact on whether a client will use pipelining or not, a server cannot prevent a client from pipelining requests, since there's nothing stopping a client from sending 10 requests in a single packet. What it does control is whether, when the client does pipeline requests, the server will process the pipelined requests in parallel or serially.